clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -386,15 +386,25 @@ private void doSend(ClientRequest clientRequest, boolean isInternalRequest, long
      */
     @Override
     public int inFlightRequestCount() {
-        return this.inFlightRequests.inFlightRequestCount();
+        return this.inFlightRequests.count();
+    }
+
+    @Override
+    public boolean hasInFlightRequests() {
```

Comment:
Seems we do have a couple tests where we could add some simple assertions for this method.